src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): Record<string, T[]> {
+  return items.reduce((grouped: GroupsMap<T>, item: T) => {
+    const itemKey = item[key] as unknown as string;
```

Comment:
Change key and key type instead of using `as`

```suggestion
    const itemKey = String(item[key])
```